### ISSUE TITLE
[BUG] Latched cvars do not give feedback when changed in the menu

### ISSUE THREAD

**Issue by bcahue** [odamex/odamex#724]
**Describe the bug**
When changing a latched cvar in the menu (For example, "Fit hit detection at grid edges" in the compatibility menu), the option will not change in the menu. However, if you restart the map, the option does indeed change, but there's no feedback to the user that that will be done.

**Build that the bug occurred in**
Odamex 10.1 Stable

**To Reproduce**
In a local game, in the menu, select Options, select Compatibility Options and try and flip "Fit hit detection at grid edges".

**Expected behavior**
Cvar should change, a message should be shown that the cvar is changed and will take effect on map restart.

**Screenshots, NetDemos, & Crash Dumps**
N/A

**Additional context**
N/A

**Comment by rakohus** [comment]
In terms of function on the local game, maybe the player should be prompted with a "Reload the Map now with changes? y/n" after leaving that menu. Not all players probably even know the console exists. Just a thought

**Comment by rakohus** [comment]
Disregard above comment, I misunderstood the issue on first reading

**Comment by bcahue** [comment]
I think changing the menu options to their pending option is the play here, but perhaps a graphic would help denote that the change will take effect next map, and not be as clunky as a message saying the same?

**Comment by electricbrass** [comment]
I agree that the menu options should change to the pending value. I think as long as the pending value does not match the current value, a message should be displayed at the bottom that says the change will not take effect until the next map. If the settings are changed back to their previous values, the message should disappear (and obviously it should go away when the next map is loaded too, presumably there'd just be a check somewhere for whether any of the settings currently will change values when they are unlatched).
